cmd/thanos/query-frontend.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package main
+
+import (
+	"time"
+
+	"github.com/alecthomas/units"
+	"github.com/cortexproject/cortex/pkg/querier/frontend"
+	"github.com/cortexproject/cortex/pkg/querier/queryrange"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/run"
+	"github.com/opentracing/opentracing-go"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/common/route"
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	v1 "github.com/thanos-io/thanos/pkg/api/queryfrontend"
+	"github.com/thanos-io/thanos/pkg/component"
+	"github.com/thanos-io/thanos/pkg/extprom"
+	"github.com/thanos-io/thanos/pkg/prober"
+	"github.com/thanos-io/thanos/pkg/queryfrontend"
+	httpserver "github.com/thanos-io/thanos/pkg/server/http"
+)
+
+type queryFrontendConfig struct {
+	http             httpConfig
+	queryRangeConfig queryRangeConfig
+
+	downstreamURL        string
+	compressResponses    bool
+	LogQueriesLongerThan time.Duration
+}
+
+type queryRangeConfig struct {
+	respCacheConfig     responseCacheConfig
+	cacheResults        bool
+	splitInterval       model.Duration
+	maxRetries          int
+	maxQueryParallelism int
+	maxQueryLength      model.Duration
+}
+
+type responseCacheConfig struct {
+	cacheMaxFreshness time.Duration
+	fifoCache         fifoCacheConfig
+}
+
+func (c *responseCacheConfig) registerFlag(cmd *kingpin.CmdClause) {
+	c.fifoCache.registerFlag(cmd)
+	cmd.Flag("query-range.response-cache-max-freshness", "Most recent allowed cacheable result, to prevent caching very recent results that might still be in flux.").
```

Comment:
I believe this should be overall general config not fifo? :thinking:

Reply:
Yes, you are right. It is a general cache config so I named it `query-range.response-cache-max-freshness`.